src/navigationAdapter.ts

```diff
@@ -230,7 +230,18 @@ function morphResponse (response: FetchResponseInterface, pushState: boolean = f
       if (errorRenderer === 'turbo') {
         renderError(html)
       } else if (errorRenderer === 'morphdom') {
-        morphHtml(html)
+        const selectorString = element.getAttribute('data-ujs-selector')
+        let selector
+
+        //  query for selectorString, if none fall back to document.body
+        if (selectorString != null) {
```

Comment:
```suggestion
        if (!response.succeeded && selectorString != null) {
```

This change would do the trick in terms of only updating if the response was not successful.